File: Study/Programming Skills/10. Monotonic Array/Go/main.go
func isMonotonic(nums []int) bool {
  inc, dec := true, true  
  for i := 1; i < len(nums); i++ {
    if nums[i] < nums[i-1] {
      inc = false
    }
    if nums[i] > nums[i-1] {
      dec = false
    }
  }
  return inc || dec
}

File: Study/Programming Skills/10. Monotonic Array/Rust/main.rs
impl Solution {
    pub fn is_monotonic(nums: Vec<i32>) -> bool {
       let mut increasing = true;
       let mut decreasing = true;

       for i in 1..nums.len() {
           if nums[i] > nums[i-1] {
               decreasing = false;
           }
           if nums[i] < nums[i-1] {
               increasing = false;
           }
       }

       increasing || decreasing
    }
}
